app/com/arpnetworking/play/ProxyClient.java

```diff
@@ -0,0 +1,112 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.play;
+
+import akka.util.ByteString;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+import org.asynchttpclient.AsyncHandler;
+import org.asynchttpclient.AsyncHttpClient;
+import org.asynchttpclient.RequestBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import play.libs.ws.WSClient;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+/**
+ * A simple proxy client.
+ *
+ * @author Brandon Arp (barp at groupon dot com)
+ */
+public class ProxyClient {
+    /**
+     * Public constructor.
+     *
+     * @param baseUrl the base url for the proxy
+     * @param client ws client to use
+     */
+    @AssistedInject
+    public ProxyClient(@Assisted final String baseUrl, final WSClient client) {
+        _baseUrl = baseUrl;
+        _client = client;
+    }
+
+    /**
+     * Proxy a request.
+     *
+     * @param path the path to proxy
+     * @param request the request
+     * @param handler a handler to execute as a callback
+     * @param <T> the type of the handler
+     */
+    public <T> void proxy(
+            final String path,
+            final play.mvc.Http.Request request,
+            final AsyncHandler<T> handler) {
+
+        final ByteString body = request.body().asBytes();
+        final URI uri = uri(path);
+        LOGGER.info(String.format("Proxy url: %s", uri));
+
+        final RequestBuilder builder = new RequestBuilder();
+        for (final Map.Entry<String, String[]> entry : request.queryString().entrySet()) {
+            for (final String val : entry.getValue()) {
+                builder.addQueryParam(entry.getKey(), val);
+            }
+        }
+
+        builder.setUrl(uri.toString());
+        builder.setMethod(request.method());
+        for (final Map.Entry<String, String[]> entry : request.headers().entrySet()) {
+            for (final String val : entry.getValue()) {
+                builder.setHeader(entry.getKey(), val);
+            }
+        }
+        if (body != null) {
+            builder.setBody(body.asByteBuffer());
+        }
+
+        final Object underlying = _client.getUnderlying();
+        if (underlying instanceof AsyncHttpClient) {
+            final AsyncHttpClient client = (AsyncHttpClient) underlying;
+            client.executeRequest(builder.build(), handler);
+        } else {
+            throw new RuntimeException("Unknown AsyncHttpClient '" + underlying.getClass().getCanonicalName() + "'");
+        }
+    }
+
+    /**
+     * Computes an aboslute URI from a relative path.
+     *
+     * @param relativePath the relative path
+     * @return the absolute URI
+     */
+    protected URI uri(final String relativePath) {
+        try {
+            return new URI(_baseUrl).resolve(relativePath);
```

Comment:
Should convert base uri from string to uri in constructor.

Reply:
^ Then this method is not necessary

Reply:
Done.